### PR TITLE
fix(components/AppHeaderBar): draft for AppHeaderBar changes

### PR DESCRIPTION
**Pull request by mtikhinskaya**
**Please check if the PR fulfills these requirements**

- [x] The commit(s) message(s) follows our
  [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md#commit-message-format)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What is the current behavior?** (You can also link to an open issue here)

image

**What is the new behavior?**

image

**Does this PR introduce a breaking change?**

- [x] Yes
- [ ] No

If this PR contains a breaking change, please describe the impact and migration
path for existing applications: ...

**Other information**:
Frontify: http://guidelines.talend.com/document/92132#/navigation-layout/app-header

ATTENTION: Adding Action and ActionDropdown in Nav will cause warning:
 `Unknown props 'activeKey', 'activeHref' on <button> tag`

### CODE REVIEW

File: packages/components/stories/Layout.js [hunk 1]
```diff
@@ -57,7 +48,46 @@ const dockedSidePanel = (<SidePanel
...
+	products: {
+		items: [
+			{
+				icon: 'talend-logo-square',
+				label: 'Portal',
+				onClick: action('onPortalClick'),
+			},
+		],
+	},
+};
```
Comment (talend-glorieux) [inline packages/components/stories/Layout.js]: API Looks neat 👍

Comment (mtikhinskaya) [inline packages/components/stories/Layout.js]: Thank you

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://316.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://316.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://316.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://316.talend.surge.sh)

File: packages/components/stories/AppHeaderBar.js [hunk 2]
```diff
@@ -54,186 +55,140 @@ const typeaheadItems = [
...
 		</div>
 	));
 
 decoratedStories
 	.addWithInfo('default', () => {
-		if (props.content[1]) {
-			delete props.content[1];
-		}
-		return <AppHeaderBar {...props} />;
+		const headerProps = Immutable.fromJS(props).toJS();
```
Comment (sdwvit) [inline packages/components/stories/AppHeaderBar.js]: is this like copying object?
_.deepClone might be more clear here

Comment (mtikhinskaya) [inline packages/components/stories/AppHeaderBar.js]: It was used in List story (https://github.com/Talend/ui/blob/master/packages/components/stories/List.js#L307) so I just reuse it. But we can discuss it I think

Comment (sdwvit) [inline packages/components/stories/AppHeaderBar.js]: it's ok then

File: packages/components/src/AppHeaderBar/AppHeaderBar.component.js [hunk 3]
```diff
@@ -1,215 +1,180 @@
...
+					{!props.logo.isFull && '|'}
+					<Brand {...props.brand} />
+					{props.env && '|'}
 				</Navbar.Brand>
 				<Navbar.Toggle />
 			</Navbar.Header>
 			<Navbar.Collapse>
-				{props.content && props.content.map(renderContent)}
+				<Nav>
+					{props.env && (<Environment {...props.env} />)}
```
Comment (jmfrancois) [inline packages/components/src/AppHeaderBar/AppHeaderBar.component.js]: if props.env is false I will have an empty Nav.
Could you please put the props.env condition on the <Nav> ?

File: packages/components/src/AppHeaderBar/AppHeaderBar.component.js [hunk 4]
```diff
@@ -1,215 +1,180 @@
...
+				</Nav>
+				<Nav pullRight>
+					|
```
Comment (jmfrancois) [inline packages/components/src/AppHeaderBar/AppHeaderBar.component.js]: Please replace all occurences of this '|' caracters from the HTML and use a before/after CSS instead

File: packages/components/src/AppHeaderBar/AppHeaderBar.component.js [hunk 5]
```diff
@@ -1,215 +1,180 @@
...
+					<Help {...props.help} />
+					<User {...props.user} />
+					|
```
Comment (jmfrancois) [inline packages/components/src/AppHeaderBar/AppHeaderBar.component.js]: same

File: packages/components/src/AppHeaderBar/AppHeaderBar.component.js [hunk 6]
```diff
@@ -1,215 +1,180 @@
...
-	buttonLabel: React.PropTypes.string,
+
+User.propTypes = {
+	id: React.PropTypes.string.isRequired,
+	items: ActionDropdown.propTypes.items,
+	name: ActionDropdown.propTypes.label,
 };
 
-export function renderTypeahead(search, index) {
+function Products({ id, items }) {
```
Comment (jmfrancois) [inline packages/components/src/AppHeaderBar/AppHeaderBar.component.js]: Please export each component so we can tests them.

File: packages/components/src/AppHeaderBar/AppHeaderBar.component.js [hunk 7]
```diff
@@ -1,215 +1,180 @@
...
+					|
+					<Products {...props.products} />
+				</Nav>
+				<Navbar.Form pullRight role="search">
+					<Search {...props.search} />
+				</Navbar.Form>
 			</Navbar.Collapse>
 		</Navbar>
 	);
 }
```
Comment (jmfrancois) [inline packages/components/src/AppHeaderBar/AppHeaderBar.component.js]: We could attach all sub components here
AppHeaderBar.Products = Products;
AppHeaderBar.Brand = Brand;
etc...

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://316.talend.surge.sh)

File: packages/components/src/AppHeaderBar/AppHeaderBar.component.js (hunk 6 above)
Comment (mtikhinskaya) [inline packages/components/src/AppHeaderBar/AppHeaderBar.component.js]: Exported as subcomponent of AppHeaderBar

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://316.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://316.talend.surge.sh)

**Comment by mtikhinskaya** [comment]
It was decided to build HeaderBar without Navbar. Please check #333

**Pull request closed without merging**